docs/examples/quickstart/test_quickstarts.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+require 'English'
+require 'fileutils'
+require 'timeout'
+
+require 'bundler'
+
+CURL_ERROR = "curl: (7) Failed to connect to localhost port 5050: Connection refused\n"
+CURL_EMPTY = "curl: (52) Empty reply from server\n"
+
+quickstarts = [
+  { lang: 'java', setup: 'make build', server: 'make run' },
+  { lang: 'nodejs', setup: 'npm i', server: 'npm start' },
+  { lang: 'python', setup: 'pip install --upgrade -r requirements.txt', server: 'python server.py' },
+  { lang: 'ruby', setup: 'bundle', server: 'bundle exec ruby server.rb' },
+  { lang: 'rust', setup: 'cargo build', server: 'cargo run' }
+]
+
+def start_server(server, user, expense_id)
+  server = spawn server
+  received = CURL_ERROR
+  while [CURL_ERROR, CURL_EMPTY].include? received
+    sleep 0.5
+    received = `curl -sSH "user: #{user}" localhost:5050/expenses/#{expense_id} 2>&1`
+  end
+  [server, received]
+end
+
+def ensure_port_5050_is_open
+  until (server = `lsof -ti :5050 2>&1`.split.first.to_i).zero?
```

Comment:
In Ruby, assignments are expressions, so the value of `x = 1` is `1`. This line is saying "keep running `lsof -ti :5050 2>&1` until parsing the first line of the output as an integer returns `0`, which will never be a pid. If it returns a non-zero int, it means there's at least one pid still listening on port 5050. `.to_i` returns `0` if it's passed something it can't reasonably parse into an integer -- in this case, an empty string b/c `lsof -ti :5050 2>&1` returned nothing b/c there's no longer anything listening on port 5050.

Not the most intuitive code, but it gets the job done and I didn't spend much effort cleaning it up.